src/core/parser.js

```diff
@@ -933,9 +933,14 @@ class Lexer {
     if (ch < /* '0' = */ 0x30 || ch > /* '9' = */ 0x39) {
       const msg = `Invalid number: ${String.fromCharCode(ch)} (charCode ${ch})`;
 
-      if (isWhiteSpace(ch) || ch === /* EOF = */ -1) {
+      if (
```

Comment:
Why not always returning 0 here whatever the next char is ?

Reply:
I'd suggest that we try that in a follow-up, rather than here, since it's impossible for me to say if that's *generally* safe.